sunbird-es-utils/src/main/java/org/sunbird/common/ElasticSearchRestHighImpl.java

```diff
@@ -0,0 +1,1033 @@
+                    + System.currentTimeMillis()
+                    + " for Type "
+                    + type
+                    + " ,Total time elapsed = "
+                    + calculateEndTime(startTime),
+                LoggerEnum.PERF_LOG);
+          }
+        };
+
+    ConnectionManager.getRestClient().indexAsync(indexRequest, listener);
+
+    return promise.future();
+  }
+
+  @Override
+  public Future<Boolean> updateData(
+      String index, String type, String identifier, Map<String, Object> data) {
+    long startTime = System.currentTimeMillis();
+    ProjectLogger.log(
+        "ElasticSearchUtilRest updateData method started at ==" + startTime + " for Type " + type,
+        LoggerEnum.PERF_LOG);
+    Promise<Boolean> promise = Futures.promise();
+    ;
+
+    if (!StringUtils.isBlank(index)
+        && !StringUtils.isBlank(type)
+        && !StringUtils.isBlank(identifier)
+        && data != null) {
+      Map<String, String> mappedIndexAndType =
+          ElasticSearchHelper.getMappedIndexAndType(index, type);
+      UpdateRequest updateRequest =
+          new UpdateRequest(
+                  mappedIndexAndType.get(JsonKey.INDEX),
+                  mappedIndexAndType.get(JsonKey.TYPE),
+                  identifier)
+              .doc(data);
+
+      ActionListener<UpdateResponse> listener =
+          new ActionListener<UpdateResponse>() {
+            @Override
+            public void onResponse(UpdateResponse updateResponse) {
+              promise.success(true);
+              if (updateResponse.getResult() == DocWriteResponse.Result.CREATED) {
+                ProjectLogger.log(
+                    "ElasticSearchUtilRest updateData  Success with upsert for index : "
+                        + index
+                        + ",type : "
+                        + type
+                        + ",identifier : "
+                        + identifier,
+                    LoggerEnum.INFO);
+              } else if (updateResponse.getResult() == DocWriteResponse.Result.UPDATED) {
+                ProjectLogger.log(
+                    "ElasticSearchUtilRest updateData  Success for index : "
+                        + index
+                        + ",type : "
+                        + type
+                        + ",identifier : "
+                        + identifier,
+                    LoggerEnum.INFO);
+              } else if (updateResponse.getResult() == DocWriteResponse.Result.NOOP) {
+                promise.success(false);
+                ProjectLogger.log(
+                    "ElasticSearchUtilRest updateData  falied for index : "
+                        + index
+                        + ",type : "
+                        + type
+                        + ",identifier : "
+                        + identifier,
+                    LoggerEnum.INFO);
+              }
+
+              long stopTime = System.currentTimeMillis();
```

Comment:
used defined method to calculate end time.

Reply:
Done